File: src-ts/Stack/static/Stack.ts
export default class Stack {
    _stack: Array<number>;
    size: number;

    constructor(size: number) {
        this._stack = [];
        this.size = size;
    }

    is_empty(): boolean {
        return !this._stack.length;
    }

    is_full(): boolean {
        return this._stack.length === this.size;
    }

    push(value: number) {
        try {
            if (this.is_full())
                throw "Stack is full. Can't push more data.";
            this._stack.push(value);
        } catch (err) {
            console.error(err)
        }
    }

    pop(): number | undefined {
        try {
            if (this.is_empty())
                throw "Stack is empty.";
            return this._stack.pop();
        } catch (err: any) {
            console.error(err)
        }
    }

    peek(): number | undefined {
        try {
            if (this.is_empty())
                throw "Stack is empty. Please, push some value";
            return this._stack[this._stack.length - 1];
        } catch (err: any) {
            console.error(err)
        }
    }

    toString(): string {
        let str = "\tHEAD -> ";
        str += [...this._stack].reverse().join(" -> ")
        str += "NULL\n";
        return str;
    }
}

File: src-py/Stack/static/Stack.py
class Stack:
    def __init__(self, size: int) -> None:
        self._stack = []
        self.size = size
        
    def __del__(self) -> None:
        self._stack.clear()
        
    def __str__(self) -> str:
        str_stack = "\tHEAD -> "
        str_stack += " -> ".join(str(e) for e in self._stack[::-1])
        str_stack += " -> NULL\n"
        return str_stack
        
    def is_empty(self) -> bool:
        return len(self._stack) == 0
    
    def is_full(self) -> bool:
        return len(self._stack) == self.size
    
    def push(self, value: int) -> None:
        try:
            if self.is_full():
                raise Exception("Stack is full. Can't push more data")
            self._stack.append(value)
        except Exception as exc:
            msg = exc.args
            print(f"Error: {msg[0]}")
        
    def pop(self) -> int:
        try:
            if self.is_empty():
                raise Exception("Stack is empty.")
            return self._stack.pop()
        except Exception as exc:
            print(exc[0])
    
    def peek(self) -> int:
        try:
            if self.is_empty():
                raise Exception("Stack is empty.")
            return self._stack[-1]
        except Exception as exc:
            print(exc[0])